lib/rubocop/cop/rspec/mixin/css_selector.rb

```diff
@@ -55,6 +55,17 @@ def common_attributes?(selector)
           attributes(selector).keys.difference(COMMON_OPTIONS).none?
         end
 
+        # @param selector [String]
+        # @return [Array<String>]
+        # @example
+        #   pseudo_classes('button:not([disabled])') # => ['not()']
+        #   pseudo_classes('a:enabled:not([valid])') # => ['enabled', 'not()']
+        def pseudo_classes(selector)
+          # Attributes must be excluded or else the colon in the `href`s URL
+          # will also be picked up as pseudo classes.
+          selector.gsub(/\[.*?\]/, '').scan(/:(.*)/).flatten.join.split(':')
```

Comment:
I wonder if `scan(/:(.*)/)` should be `scan(/:(.*?)/)` to avoid matching `:enabled:not` but only `:enabled`. I guess I am not sure what this code does 😅

Perhaps you could split this line into two or three, using descriptive local variable names?

Reply:
I separated the method chain and added a comment. What do you think?
https://github.com/rubocop/rubocop-rspec/blob/2fa8f2aea0c77eb76c100ffcd1feaf71ab4d6430/lib/rubocop/cop/rspec/mixin/css_selector.rb#L63-L72